core/src/streams.rs

```diff
@@ -269,6 +271,44 @@ impl<'gc> NetStream<'gc> {
         self.0.write(gc_context).avm_object = Some(avm_object);
     }
 
+    /// Initialize the `NetStream` buffer to accept new source data.
+    ///
+    /// This must be done once per source change and should ideally be done
+    /// immediately before the first `load_buffer` call for a particular source
+    /// file.
+    ///
+    /// Externally visible AVM state must not be initialized here - i.e. the
+    /// AS3 `client` doesn't go away because you played a new video file.
+    pub fn init_buffer(self, context: &mut UpdateContext<'_, 'gc>) {
```

Comment:
Isn't this actually a "clear" or a "reset" operation? :thinking:

Reply:
Yes